.github/workflows/python-ci.yml

```diff
@@ -34,5 +34,29 @@ jobs:
       - name: Install python dev dependencies
         run: uv pip install -r context/requirements-dev.txt
 
+      - name: Install and register Jupyter kernel
+        run: |
+          uv pip install ipykernel
+          python -m ipykernel install --user --name python3
+          python - <<'PY'
+          import json, subprocess
+          try:
+              out = subprocess.check_output(["jupyter", "kernelspec", "list", "--json"])
+              specs = json.loads(out.decode())["kernelspecs"].keys()
+              print("Kernelspecs:", ", ".join(specs))
+          except Exception as e:
+              print("Could not list kernelspecs:", e)
+          PY
+
       - name: Run test script
         run: etc/test/test-python.sh
+
+      - name: Generate figure SVGs
+        run: python context/app/figure/build_figure.py
```

Comment:
Given timing and resources it doesn't make sense, but it would have been interesting to create a custom github action so we could keep some of the figure building logic out of this repo.